### PR TITLE
Adding slurm exclusive kwarg

### PR DESCRIPTION
**Pull request by yadudoc**
This PR adds a new KWarg "exclusive" to the SlurmProvider. Htex and Exex assume nodes are exclusive and that assumption is exposed more clearly via setting the default to True.

### CODE REVIEW

**Comment by annawoodard** [comment]
I'm worried this is a bit too aggressive. The point of setting ntasks-per-node in the submit script was just to make sure not all processes were restricted to a single core. I prefer to just set `ntasks-per-node` in the submit script appropriately rather than always default to exclusive (which can drastically increase queue times, there may be cases where you don't mind sharing even though you want one manager per node). For ipp `ntasks-per-node=tasks_per_node`, for htex `ntasks-per-node=cores_per_node / cores_per_worker + 1` (for the manager), no?

**Comment by annawoodard** [comment]
Note this PR addresses #660

**Comment by yadudoc** [comment]
The general assumption is that we provision the whole node and do the partitioning. These changes make those assumptions explicit. Specifying the # of workers in this situation is straightforward. I think that this is the more common/preferred mode. 

The core issue is that HTEX assumes control of the whole node, so it starts workers with the right # of cores/mem. In this model -ntasks-per-node is not known ahead of time. So you'll have to set it manually every time partitions are switched.

File: parsl/providers/slurm/slurm.py [hunk 1]
```diff
@@ -59,6 +59,9 @@ class SlurmProvider(ClusterProvider, RepresentationMixin):
         String to prepend to the #SBATCH blocks in the submit script to the scheduler.
     worker_init : str
         Command to be run before starting a worker, such as 'module load Anaconda; source activate env'.
+    exclusive : bool (Default = True)
+        Requests nodes which are not shared with other running jobs. Executors such as HighThroughputExecutor
+        and ExtremeScaleExecutor assume that the nodes are exclusive.
```
Comment (ZhuozhaoLi) [inline parsl/providers/slurm/slurm.py]: Based on our discussion, we should remove htex and exex from the docstring to avoid confusion.

**Pull request merged**